setup/index.ts

```diff
@@ -1,11 +1,31 @@
-import { askSessionToken } from './prompt';
-import { get as getEnvValue, set as setEnvValue } from './env';
+import {
+  /* askSessionToken, */
+
+  askIOBackendHost,
+  askIOBackendBasePath
+} from "./prompt";
+import { get as getEnvValue, set as setEnvValue, EnvKey } from "../lib/env";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { task } from "fp-ts/lib/Task";
+
+const promptIfNot = (
+  key: EnvKey,
+  promptFn: () => TaskEither<Error, string>
```

Comment:
while it's ok to return monadic structures as output (option, eithers, ...) try to use plain values when possible as arguments (otherwise you're this forcing the caller to "lift" the arguments structures). in this case a plain async method would have been ok (but as we said, try to not pass callbacks :-)

what you're plrobably try to accomplish here can be expresses more explicitely (without hiding the fact that you're changing process.env global object and without modifying it): <--never mutate global objects, try not to mutale local objects:

```ts
const ask = (key: string, message: string, env = process.env) =>
  fromIO(io.of(fromNullable(env[key]))).chain(() =>
    tryCatch(async () => singlePrompt({ message }), toError)
  );

const config = sequenceS(T.taskEither)({
  bar: ask("bar", "prompt message.."),
  foo: ask("foo", "prompt message..")
});

config.map(console.log).run();

// result ->
// { 
//     bar: ' ... value from singlePrompt...',
//     foo: 'value from singlePrompt...' 
// }
```